src/validator.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  BaseMetric,
+  baseMetrics,
+  BaseMetricValue,
+  baseMetricValues
+} from './models';
+import { humanizeBaseMetric, humanizeBaseMetricValue } from './humanizer';
+import { parseMetricsAsMap, parseVector, parseVersion } from './parser';
+
+export const validateVersion = (versionStr: string | null): void => {
+  if (!versionStr) {
+    throw new Error(
+      'Invalid CVSS string. Example: CVSS:3.0/AV:A/AC:H/PR:H/UI:R/S:U/C:N/I:N/A:L'
+    );
+  }
+
+  if (versionStr !== '3.0' && versionStr !== '3.1') {
+    throw new Error(
+      `Unsupported CVSS version: ${versionStr}. Only 3.0 and 3.1 are supported`
+    );
+  }
+};
+
+const validateVector = (vectorStr: string | null): string => {
+  if (!vectorStr || vectorStr.includes('//')) {
+    throw new Error(
+      'Invalid CVSS string. Example: CVSS:3.0/AV:A/AC:H/PR:H/UI:R/S:U/C:N/I:N/A:L'
+    );
+  }
+
+  return vectorStr;
+};
```

Comment:
The return value isn't used
```suggestion
const validateVector = (vectorStr: string | null): void => {
  if (!vectorStr || vectorStr.includes('//')) {
    throw new Error(
      'Invalid CVSS string. Example: CVSS:3.0/AV:A/AC:H/PR:H/UI:R/S:U/C:N/I:N/A:L'
    );
  }
};
```